src/search/parser/syntax_analyzer.h

```diff
@@ -0,0 +1,11 @@
+#ifndef PARSER_SYNTAX_ANALYZER_H
+#define PARSER_SYNTAX_ANALYZER_H
+
+#include "abstract_syntax_tree.h"
+
+namespace parser {
+class TokenStream;
+
+ASTNodePtr parse_ast(TokenStream &tokens);
```

Comment:
Now `extern` and renamed to `parse`. The argument `TokenStream` indicates what is parsed by this function.
 I prefer the name `SyntaxAnalyzer` over `Parser`, because the filename indicates clearly what is analyzed (even if you do not know the conventation that the Parser does the syntax analysis) and it cannot be confused with our old parsing framework.